src/app/tools/point_shapes.h

```diff
@@ -208,7 +260,7 @@ class BrushPointShape : public PointShape {
 
     ink->prepareForPointShape(loop, m_firstPoint, x, y);
 
-    for (auto scanline : *m_compressedImage) {
+    for (auto scanline : *m_compressedImages[int(pt.symmetry)]) {
```

Comment:
I would prefer something like:
```suggestion
    for (auto scanline : getCompressedImage(pt.symmetry)) {
```
And `getCompressedImage()` could create the compressed brush when it's needed (instead of `setSymmetricCommpressedImages()` where we might be creating and re-creating several flipped version of brushes that will not be used in each specific `transformPoint()`).

In this case the code when if (m_lastBrush != brush) should reset all cached compressed images:
```c++
if (m_lastBrush = brush) {
  m_lastBrush = brush;
  m_compressedImages.fill(nullptr);
}
```
And change `m_compressedImages` type to:
```c++
std::array<std::shared_ptr<CompressedImage>, 4> m_compressedImages;
```